internal/provider/meta_test.go

```diff
@@ -671,12 +673,32 @@ func TestNewProviderMeta(t *testing.T) {
 					},
 				},
 			},
+			env: map[string]string{
```

Comment:
nit: I confirmed this with commenting out the `env` in this test and the test passes so we're definitely good here, but curious if it's worth it to add a test case for the default behavior where no env is passed

Reply:
I think we are testing that in several test cases above this one